banzai/frames.py

```diff
@@ -21,6 +19,7 @@ def __init__(self, hdu_list: list, file_path: str, frame_id: int = None, hdu_ord
         self.instrument = None
         self.frame_id = frame_id
         self.hdu_order = hdu_order
+        self._hdu_keys = {hdu.name: i for i, hdu in enumerate(hdu_list)}
```

Comment:
The name `_hdu_keys` threw me for a bit of a loop. `_extension_indices`, `_hdu_indices` maybe? Something to reflect that we're storing a mapping of HDU name to index.